crates/services/executor/src/executor.rs

```diff
@@ -689,41 +615,32 @@ where
                 coinbase_contract_id,
                 gas_price,
                 execution_data,
-                execution_kind,
                 &mut tx_st_transaction,
             )?;
             tx_st_transaction.commit()?;
             tx
         };
 
-        block.transactions.push(tx);
+        block.transactions.push(tx.clone());
         execution_data.tx_count = tx_count
             .checked_add(1)
             .ok_or(ExecutorError::TooManyTransactions)?;
 
-        Ok(())
+        Ok(tx)
     }
 
     #[tracing::instrument(skip_all)]
-    fn validate_block(
-        mut self,
-        component: PartialBlockComponent<OnceTransactionsSource>,
-    ) -> ExecutorResult<ExecutionData> {
-        let block_gas_limit = self.consensus_params.block_gas_limit();
+    fn validate_block(mut self, block: &Block) -> ExecutorResult<ExecutionData> {
         let mut data = ExecutionData::new();
 
-        let block = component.empty_block;
-        let source = component.transactions_source;
-        let gas_price = component.gas_price;
-        let coinbase_contract_id = component.coinbase_contract_id;
-        let block_height = *block.header.height();
-        let execution_kind = ExecutionKind::Validation;
+        let mut partial_block = PartialFuelBlock::from(block.clone());
+        let transactions = core::mem::take(&mut partial_block.transactions);
```

Comment:
It would be nice if you avoided cloning of the block here=) You don't need to modify `transactions`, so you can work with the reference